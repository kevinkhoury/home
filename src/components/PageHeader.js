import React from 'react'
import PropTypes from 'prop-types'

const PageHeader = ({ header, subheader, missionStatement }) => (
  <header className="page-head">
    <h1 className="page-head-title">{header}</h1>
    {!!subheader && <p className="page-head-description">{subheader}</p>}
    {!!missionStatement && (
      <p className="page-head-mission-statement">{missionStatement}</p>
    )}
  </header>
)

PageHeader.propTypes = {
  header: PropTypes.string.isRequired,
  subheader: PropTypes.string,
  missionStatement: PropTypes.string,
}

export default PageHeader
