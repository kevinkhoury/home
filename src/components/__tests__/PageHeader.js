import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import PageHeader from '../PageHeader'

const DEFAULT_PROPS = {
  header: 'header',
  subheader: 'subheader',
  missionStatement: 'missionStatement',
}

describe('PageHeader', () => {
  it('snapshot renders properly', () => {
    const tree = renderer.create(<PageHeader {...DEFAULT_PROPS} />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
  it('renders all props as text', () => {
    render(<PageHeader {...DEFAULT_PROPS} />)
    const header = screen.getByText('header')
    const subheader = screen.getByText('header')
    const missionStatement = screen.getByText('missionStatement')

    expect(header).toBeTruthy()
    expect(subheader).toBeTruthy()
    expect(missionStatement).toBeTruthy()
  })
  it('renders only header if that is all that is provided', () => {
    render(<PageHeader header="header" />)
    const header = screen.getByText('header')
    expect(header).toBeTruthy()
    expect(screen.queryByText('subheader')).toBeNull()
    expect(screen.queryByText('missionStatement')).toBeNull()
  })
})
