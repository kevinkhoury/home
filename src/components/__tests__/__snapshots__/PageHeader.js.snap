// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader snapshot renders properly 1`] = `
<header
  className="page-head"
>
  <h1
    className="page-head-title"
  >
    header
  </h1>
  <p
    className="page-head-description"
  >
    subheader
  </p>
  <p
    className="page-head-mission-statement"
  >
    missionStatement
  </p>
</header>
`;
